projects/noloan/app/src/main/java/hasadna/noloan/DbMessages.java

```diff
@@ -0,0 +1,104 @@
+package hasadna.noloan;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import hasadna.noloan.protobuf.SmsProto.SmsMessage;
+
+// Simple singleton to hold the suggestions list
+public class DbMessages {
+  private static DbMessages instance;
+  private List<SmsMessage> spam;
+  private List<SmsMessage> suggestions;
+
+  MessagesListener spamListener;
+  MessagesListener suggestionsListener;
+
+  public static DbMessages getInstance() {
+    if (instance == null) {
+      instance = new DbMessages();
+    }
+    return instance;
+  }
+
+  public DbMessages() {
+    spam = new ArrayList<>();
+    suggestions = new ArrayList<>();
+  }
+
+  public void init(List<SmsMessage> spam, List<SmsMessage> suggestions) {
+    this.spam = spam;
+    this.suggestions = suggestions;
+  }
+
+  public List<SmsMessage> getSpam() {
+    return spam;
+  }
+
+  public List<SmsMessage> getSuggestions() {
+    return suggestions;
+  }
+
+  public void addSpam(SmsMessage spam) {
+    this.spam.add(spam);
+
+    if (spamListener != null) {
+      spamListener.messageAdded();
+    }
+  }
+
+  public void spamModified(SmsMessage spam) {
+    int index = this.spam.indexOf(spam);
+    this.spam.set(index, spam);
+
+    if (spamListener != null) spamListener.messageModified(index);
+  }
+
+  public void spamRemove(SmsMessage spam) {
+    int index = this.spam.indexOf(spam);
+
+    this.spam.remove(spam);
+
+    if (spamListener != null) spamListener.messageRemoved(index);
+  }
+
+  public void addSuggestion(SmsMessage suggestedSpam) {
+    suggestions.add(suggestedSpam);
+
+    if (suggestionsListener != null) {
+      suggestionsListener.messageAdded();
+    }
+  }
+
+  public void suggestionsModified(SmsMessage suggestedSpam) {
+    int index = suggestions.indexOf(suggestedSpam);
```

Comment:
If we can't modify, please remove this functionality, especially since it might not work properly when using `indexOf` of the modified `SmsMessage`.